packages/hardhat-core/src/internal/hardhat-network/provider/vm/dual.ts

```diff
@@ -610,91 +927,108 @@ function assertEqualRunTxResults(
+    differences.push("length");
   }
 
   for (let logIdx = 0; logIdx < ethereumJSLogs.length; ++logIdx) {
     if (!ethereumJSLogs[logIdx][0].equals(rethnetLogs[logIdx][0])) {
       console.trace(
-        `Different log[${logIdx}] address: ${ethereumJSLogs[logIdx][0]} !== ${rethnetLogs[logIdx][0]}`
+        `Different log[${logIdx}] address: ${ethereumJSLogs[logIdx][0]} (ethereumjs) !== ${rethnetLogs[logIdx][0]} (rethnet)`
       );
-      throw new Error("Different log address");
+      differences.push("address");
     }
 
     const ethereumJSTopics = ethereumJSLogs[logIdx][1];
     const rethnetTopics = rethnetLogs[logIdx][1];
     if (ethereumJSTopics.length !== rethnetTopics.length) {
       console.trace(
-        `Different log[${logIdx}] topics length: ${ethereumJSTopics.length} !== ${rethnetTopics.length}`
+        `Different log[${logIdx}] topics length: ${ethereumJSTopics.length} (ethereumjs) !== ${rethnetTopics.length} (rethnet)`
       );
-      throw new Error("Different log topics length");
+      differences.push("topics length");
     }
 
     for (let topicIdx = 0; topicIdx < ethereumJSTopics.length; ++topicIdx) {
       if (!ethereumJSTopics[topicIdx].equals(rethnetTopics[topicIdx])) {
         console.trace(
-          `Different log[${logIdx}] topic[${topicIdx}]: ${ethereumJSTopics[topicIdx]} !== ${rethnetTopics[topicIdx]}`
+          `Different log[${logIdx}] topic[${topicIdx}]: ${ethereumJSTopics[topicIdx]} (ethereumjs) !== ${rethnetTopics[topicIdx]} (rethnet)`
         );
-        throw new Error("Different log topic");
+        differences.push("topic");
       }
     }
 
     if (!ethereumJSLogs[logIdx][2].equals(rethnetLogs[logIdx][2])) {
       console.trace(
-        `Different log[${logIdx}] data: ${ethereumJSLogs[logIdx][2]} !== ${rethnetLogs[logIdx][2]}`
+        `Different log[${logIdx}] data: ${ethereumJSLogs[logIdx][2]} (ethereumjs) !== ${rethnetLogs[logIdx][2]} (rethnet)`
       );
-      throw new Error("Different log data");
+      differences.push("data");
     }
   }
+
+  if (differences.length !== 0) {
+    throw new Error(`Different log fields: ${differences}`);
+  }
 }
 
 function assertEqualAccounts(
   address: Address,
   ethereumJSAccount: Account,
   rethnetAccount: Account
 ) {
+  const differences: string[] = [];
+
   if (ethereumJSAccount.balance !== rethnetAccount.balance) {
     console.trace(`Account: ${address}`);
     console.trace(
-      `Different balance: ${ethereumJSAccount.balance} !== ${rethnetAccount.balance}`
-    );
-    throw new Error("Different balance");
-  }
-
-  if (!ethereumJSAccount.codeHash.equals(rethnetAccount.codeHash)) {
-    console.trace(
-      `Different codeHash: ${ethereumJSAccount.codeHash} !== ${rethnetAccount.codeHash}`
+      `Different balance: ${ethereumJSAccount.balance} (ethereumjs) !== ${rethnetAccount.balance} (rethnet)`
     );
-    throw new Error("Different codeHash");
+    differences.push("balance");
   }
+  const inGetBalanceContext =
```

Comment:
This is sort of a hack but I thought potentially a reasonable one. This is determining whether this check is running within the context of an `eth_getBalance` call. If it is, it skips all of the account checks except for balance, since in this case the user really only cares about the balance, not any of the other fields.